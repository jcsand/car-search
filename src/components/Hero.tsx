/** @jsxImportSource @emotion/react */
import React from "react";
import { color, ColorProps } from "styled-system";
import styled from "@emotion/styled";

import { Search } from "@components/Search";

import checkmarkIcon from "@assets/checkmark.svg";

// TODO: Import properly
const HERO_BACKGROUND_IMAGE =
  "https://cdn2.rcstatic.com/com.rentalcars.185492029745.eu-west-1.web.prod.static-live/images/landing-pages/home/background-large.jpg";

const StyledHero = styled.div<ColorProps>`
  height: min(calc(-268px + 100vh), 596px);
  min-height: 290px;
  background-image: url(${HERO_BACKGROUND_IMAGE});
  background-position: 50%;
  background-size: cover;

  ${color}
`;

const HeroGradientOverlay = styled.div`
  height: inherit;
  background: linear-gradient(#1879ca, rgb(24 121 202 / 0%));
`;

const HeaderFlexContainer = styled.div`
  display: flex;
  flex-direction: column;
  max-width: 1142px;
  margin: 0 auto;
  padding: 0 calc(4px * 4);
`;

const HeroHeader = styled.h2`
  padding-top: calc(4px * 4);
  padding-bottom: 6px;
  font-size: 40px;
  font-weight: 700;
  line-height: 52px;
`;

const HeroList = styled.ul`
  display: flex;
  flex-wrap: wrap;
  align-items: flex-start;
  padding-right: calc(4px * 4);
`;

const HeroListItem = styled.li`
  margin-right: calc(4px * 4);
  font-size: 20px;
  line-height: 28px;
  list-style: none;

  &::before {
    content: "";
    display: inline-block;
    width: calc(4px * 6);
    height: calc(4px * 5);
    margin-top: calc(4px * 3);
    margin-right: calc(4px * 2);
    background: url("${checkmarkIcon}");
    background-position: center;
    background-size: cover;
  }
`;

export const Hero: React.FC = () => (
  <StyledHero color="foreground" bg="background">
    <HeroGradientOverlay>
      <HeaderFlexContainer>
        <HeroHeader>Car Hire â€“ Search, Compare &amp; Save</HeroHeader>
        <HeroList>
          <HeroListItem>Free cancellations on most bookings</HeroListItem>
          <HeroListItem>60,000+ locations</HeroListItem>
          <HeroListItem>Customer support in 40+ languages</HeroListItem>
        </HeroList>
        <Search />
      </HeaderFlexContainer>
    </HeroGradientOverlay>
  </StyledHero>
);
