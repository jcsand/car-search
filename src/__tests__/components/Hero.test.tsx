import React from "react";
import { render, waitFor, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { mockRequests } from "../__helpers__";
import { Hero } from "@@components/Hero";

describe("Hero component", () => {
  mockRequests();

  it("renders matching snapshot", async () => {
    const rendered = render(<Hero />);
    expect(rendered.baseElement).toMatchSnapshot();
  });

  it("has heading", async () => {
    render(<Hero />);

    await waitFor(() => screen.getByRole("heading"));

    expect(screen.getByRole("heading")).toHaveTextContent(
      "Car Hire â€“ Search, Compare & Save"
    );
  });

  it("has USP list", async () => {
    render(<Hero />);

    const element = await waitFor(() =>
      screen.getByText("Free cancellations on most bookings")
    );

    expect(element).toBeInstanceOf(HTMLLIElement);
  });
});
