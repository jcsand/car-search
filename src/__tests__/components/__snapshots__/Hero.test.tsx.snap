// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component renders matching snapshot 1`] = `
<body>
  <div>
    <div
      class="css-al1uuz"
      color="foreground"
    >
      <div
        class="css-3mc4v6"
      >
        <div
          class="css-cevxc4"
        >
          <h2
            class="css-76v7hz"
          >
            Car Hire â€“ Search, Compare & Save
          </h2>
          <ul
            class="css-1gzi17z"
          >
            <li
              class="css-tasv40"
            >
              Free cancellations on most bookings
            </li>
            <li
              class="css-tasv40"
            >
              60,000+ locations
            </li>
            <li
              class="css-tasv40"
            >
              Customer support in 40+ languages
            </li>
          </ul>
          <div
            class="css-mv11ig"
          >
            <div
              class="css-oz4kv0"
            >
              <input
                aria-label="Pick-up Location"
                class="css-ls0vk8"
                placeholder="Pick-up Location"
                value=""
              />
            </div>
            <button
              class="css-o225nb"
            >
              Search
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
